myhoard/restore_coordinator.py

```diff
@@ -419,7 +419,12 @@ def restore_basebackup(self) -> None:
             free_memory_percentage=self.free_memory_percentage,
         )
         try:
-            self.basebackup_restore_operation.restore_backup()
+            try:
+                self.basebackup_restore_operation.restore_backup()
+            except DiskFullError:
+                self.stats.increase("myhoard.disk_full_errors")
+                self.update_state(phase=self.Phase.failed)
+                raise
```

Comment:
No, this is only if we try several times and reached the max try. And because we'll make it fail from the first time, it won't happen.

Reply:
So, I had another look at it. If you re-raise here, the exception block below gets called and if it fails enough times, it will still mark it as broken because `self.update_state(phase=self.Phase.failed_basebackup)` will get called and this will mark it as broken in the controller. So, at least to me, it seems like you need to re-catch the exception in the outer `try: .. catch:` block as well.

Reply:
Hmmm...

The first time it'll fail, it'll mark the state as `Phase.failed`.

Then, in `restore_coordinator.py:349`, we'll have:
```python
                if self.phase in {self.Phase.completed, self.Phase.failed, self.Phase.failed_basebackup}:
                    break
```

which will end the running loop of the restore coordinator.

So, the implementation should be correct as it is now.

Reply:
OK, I also see that we're not re-starting the restore_coordinator in any of the cases so we should be good. Thanks for double-checking!